src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/ApprovedApprenticeship.cs

```diff
@@ -14,6 +15,27 @@ public class ApprovedApprenticeship : Apprenticeship
         public bool HasHadDataLockSuccess { get; set; }
         public byte? PendingUpdateOriginator { get; set; }
 
+        public ApprenticeshipStatus Status
+        {
+            get
+            {
+                switch(PaymentStatus)
+                {
+                    case PaymentStatus.Active:
+                        return DateTime.UtcNow < StartDate
+                            ? ApprenticeshipStatus.WaitingToStart
+                            : ApprenticeshipStatus.Live;
+                    case PaymentStatus.Withdrawn:
+                        return ApprenticeshipStatus.Stopped;
+                    case PaymentStatus.Paused:
+                        return ApprenticeshipStatus.Paused;
+                    case PaymentStatus.Completed:
+                        return ApprenticeshipStatus.Completed;
+                    default:
+                        throw new InvalidOperationException($"Unable to determine Status of Apprenticeship {Id}");
```

Comment:
Just out of interest do we want to throw an exception in a property getter (it's not very usual)? if the payment status is extended and an update occurs, then this reading of the apprenticeship record will continually fail. Any thoughts?

Reply:
Returning unknown and added unit to cover the mapping between PaymentStatus and apprenticeship status.